apiary.apib

```diff
@@ -5738,3 +5738,61 @@ From now on you won't be able to sign in to the **Direct Access** for specific p
                     "outBytesUncompressed": 0
                 }
             }
+
+# Group Development Branches
+
+**EXPERIMENTAL**
+
+Development branches allow you to prepare changes to configuration and tables in isolation. When ready, you can replace the production with the development branch.
+
+## Branches [/v2/storage/dev-branches/]
+
+### Create branch [POST]
+
++ Parameters
+    + name (string) - name of the branch
+
++ Request (application/x-www-form-urlencoded)
+
+    + Headers
+
+            X-StorageApi-Token: your_token
+
++ Response 202
+
+    + Body
+
+        {
+            "id": 347,
+            "status": "success",
+            "url": "https:\/\/connection.keboola.com\/\/v2\/storage\/jobs\/347",
+            "tableId": null,
+            "operationName": "devBranchCreate",
+            "operationParams": {
+                "queue": "main",
+                "values": {
+                    "name": "Refactor DB extractors"
+                }
+            },
+            "createdTime": "2020-09-05T15:29:45+0200",
+            "startTime": "2020-09-05T15:29:45+0200",
+            "endTime": "2020-09-05T15:29:45+0200",
+            "runId": "955",
+            "results": {
+                "id": 12,
+                "name": "Refactor DB extractors",
+                "created": "2020-09-05T15:29:45+0200"
+            },
```

Comment:
```suggestion
            "results": null,
```

Reply:
Tohle jsem udělal schválně. Jinak totiž není jasné, co ti ten job vrátí v results. Normálně tě to moc nezajímá. Ale tady je to to jediné co tě zajímá. 

Nikam jinam se ten response dát nedá... 

Nepřijde mi to moc matoucí...

Reply:
Tady je X míst, kde to je s waiting a ty do toho zavádíš nějakou novotu :)
Pokud myslíš že je potřeba popsat ten result, což asi dává do budoucna smysl to mít v té dokumentaci tak to přiděj do https://keboola.docs.apiary.io/#reference/jobs a tam napiš že job tohodle typu vrací tuhle response.
Podpořit to můžeš i tím že do response `[GET] /v2/storage/jobs` přidáš tu reponse. Tady ale by být neměla.

Reply:
Nesouhlasím :) 

Tohle je prostě jiný typ API, než ty které tam byly doteď. Ty ostatní byly, že něco chceš, ono ti to řekne, že to začlo dělat a výsledek nepotřebuješ, stačí, že je to success/error (třeba async upload). Tady dokud nedostaneš výsledek, tak je ti to celé k ničemu, protože nevíš IDčko té věci...

Reply:
A je blbost tu "zajímavou" response schovávat mezi joby...

Reply:
Mj. mi přijde, že ti to vrátí waiting bez výsledku prakticky jen "náhodou". Kdyby byl na konci toho callu sleep(1) tak už ti to vrátí success třeba... :) Prostě mi nepřijde, že by byl ten rozdíl tak zásadní...